syntax = "proto3";

package grpc.test;

service Greeter {
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc SayGoodbye (GoodbyeRequest) returns (GoodbyeReply) {}
  rpc Multiply (MultiplicationRequest) returns (MultiplicationReply) {}
}

message HelloRequest {
  string name = 1;
}

message HelloReply {
  string message = 1;
}

message GoodbyeRequest {
  string name = 1;
}

message GoodbyeReply {
  string message = 1;
}

message MultiplicationRequest {
  int32 factorA = 1;
  int32 factorB = 2;
}

message MultiplicationReply {
  int32 result = 1;
}
